### DISCUSSION TITLE
Can hs.chooser make choices unselectable?

### DISCUSSION THREAD

**Discussion by unphased** [Hammerspoon/hammerspoon#3676] [Q&A] [answered]
I have figured out how to set hs.chooser choices to be `valid = false` and therefore can prevent picking them by defining an `invalidCallback` on the chooser. I have also got `styledtext` to work so I can modify at least the text color of the invalid choices, which is useful.

However, I would like it if I can go one step further to make it so my arrow key can skip over the invalid choice! Is this possible?

**Comment by latenitefilms** [comment]
As a workaround you could just trigger a down arrow via keyStroke if the invalid item is highlighted?

**Reply by unphased** [reply]
I think that's probably a good workaround but it would be awkward since it would make a disabled item work as a barrier to key navigation when navigating back up! would need to listen to key events and track that state, which is going down a treacherous path. 

I will leave it alone, the chooser works well enough for what it needs to do and I don't need to push it this hard. i believe if i really cared to do something fancy with hammerspoon i should opt for web views which seem to be a thing.

**Reply by latenitefilms** [reply, marked as answer]
Yes, every user interface in [CommandPost](https://commandpost.io) (which is a fork of Hammerspoon) is built using webviews - they're very powerful:

<img width="1000" alt="image" src="https://github.com/user-attachments/assets/433f2b18-48ee-49b7-8ef1-03f820f69bb3">

**Reply by unphased** [reply]
I'm marking this as an answer because it 100% convinces me that the path to go down for fully customized ui is the web view and not the chooser.
